### PR TITLE
Add a release process

### PR DESCRIPTION
**Pull request by bilelmoussaoui**

### CODE REVIEW

File: process.md [hunk 1]
```diff
@@ -0,0 +1,14 @@
+## Release process
+- Create a new branch for the release `$version`
+- Switch to using released versions of the dependencies
```
Comment (sdroege) [inline process.md:3]:
```suggestion
- Check that the gir and gir-files submodules are up to date
- Switch to using released versions of the dependencies
```

File: process.md [hunk 2]
```diff
@@ -0,0 +1,14 @@
+## Release process
+- Create a new branch for the release `$version`
```
Comment (sdroege) [inline process.md:2]: Also for the `gir` and `gir-files` repos

File: process.md (hunk 1 above)
Comment (sdroege) [reply process.md:3]:
```suggestion
- Switch to using released versions of the dependencies
  - keep the git urls and added the branch and keep `path` for the local ones
```

File: process.md [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+- Run `cargo publish`
+- Create a new release from Github named `$version` 
+- Add the new $version to `.github/workflows/docs.yml` for the `main` branch
+- Update `/latest/stable` link on `gh-pages` (sadly still manual)
+    - `git clone $repo -b gh-pages --depth=1`
+    - `cd $repo`
+    - `unlink ./stable/latest/docs`
+    - `cd ./stable/latest`
+    - `ln -sf ../$version/docs ./docs`
+- 🎉
```
Comment (sdroege) [inline process.md:14]:
```suggestion
- Update versions in master to the next one
- 🎉
```

**Comment by GuillaumeGomez** [comment]
So just for everyone's information, I used the following commands to update the versions:

```shell
# "Normal" cargo files:
$ find . -name Cargo.toml | grep -v gir | xargs sed -i -E s/'\{ git \=.+\}'/'"0.17"'/
# sys cargo files:
$ find . -name Cargo.toml | grep -v gir | xargs sed -i -E s/'\git \=.+\"'/'version \= "0.17"'/
# For inline deps:
$ find . -name Cargo.toml | grep -v gir | xargs sed -i -E s/'\{ path \='/'\{ version = "0.17", path ='/
```

And for the `master` branch:

```shell
$ find . -name Cargo.toml | grep -v gir | xargs sed -i -E s/'version \= "0.17.+'/'version \= "0.18.0"'/
```

Of course, `0.17` and they `0.18` version numbers have to be updated accordingly.

**Comment by sdroege** [comment]
Should this maybe be merged? :)

**Comment by bilelmoussaoui** [comment]
> Should this maybe be merged? :)

Yes, should be good to go

**Pull request merged**